### PR TITLE
Add vm.network to driver create_vm_file

### PR DESCRIPTION
**Pull request by afiune**
As a user I would like to be able to customize my network connectivity by passing though some machine_options so I can create private and other kind of internal networks.

Perhaps more functionalities could be added.

### CODE REVIEW

**Comment by afiune** [comment]
@jkeiser Thoughts?

**Comment by tyler-ball** [comment]
I really wish we had tests or a README for this... <le_sigh/>

File: lib/chef/provisioning/vagrant_driver/driver.rb [hunk 1]
```diff
@@ -222,7 +222,11 @@ def create_vm_file(action_handler, vm_name, vm_file_path, machine_options)
         merged_vagrant_options = Cheffish::MergedConfig.new(machine_options[:vagrant_options], merged_vagrant_options)
       end
       merged_vagrant_options.each_pair do |key, value|
-        vm_file_content << "    config.#{key} = #{value.inspect}\n"
+        if key == 'vm.network'
+          vm_file_content << "    config.#{key}(#{value})\n"
```
Comment (tyler-ball) [inline lib/chef/provisioning/vagrant_driver/driver.rb]: Will this support something like `"forwarded_port", guest: 80, host: 8080`?

**Comment by afiune** [comment]
@tyler-ball yup! :smile:

You would do:
```
machine_options = { 
  'vagrant_options' => { 
    'vm.hostname' => "my-hostname",
    'vm.box' => "opscode-ubuntu-14.04",
    'vm.network' => ":private_network, :ip => '33.33.33.10'",
    'vm.network' => ":forwarded_port, guest: 80, host: 8080"
  }
}

with_machine_options(machine_options)

```

This is actually @eeyun 's code.. We agreed that the driver needs a little more love but at the moment this quick fix will unlock us.. Right Ian? (help me! :smile:)

**Comment by eeyun** [comment]
Yep, thats essentially the monkey-patch I had within the delivery-cluster cookbook. Worked fine. Its kind of an ugly hack but it works! :+1: Long term I echo the sentiment that the driver needs some serious love.

**Comment by eeyun** [comment]
If this gets merged my abstraction for chef-provisioning-vagrant within delivery-cluster can get simplified a bit.

**Comment by stephenlauck** [comment]
:+1:

**Comment by tyler-ball** [comment]
Lets get it merged so your cookbook can be simplified and do some cleanup further down the line.  Maybe we just allow people to specify their own ERB file?

**Pull request merged**